test/e2e/lib/components/abstract-editor-component.js

```diff
@@ -50,4 +70,20 @@ export default class AbstractEditorComponent extends AsyncBaseContainer {
 	async runInCanvas( cb ) {
 		return await cb();
 	}
+
+	async dismissNuxWelcomeModalIfDisplayed() {
+		const element = await this.driver
+			.findElement( By.css( '.wpcom-editor-welcome-tour-frame' ) )
+			.catch( () => false );
+
+		const isNuxWelcomeDialogDisplayed = element && ( await element.isDisplayed() );
+
+		if ( isNuxWelcomeDialogDisplayed ) {
+			// Click the 'Skip' button to get rid of it.
+			await driverHelper.clickWhenClickable(
+				this.driver,
+				By.css( '.components-card-footer button.is-tertiary' )
+			);
+		}
```

Comment:
I think it would be safer to use the _waiter_ helper here with at least 1s timeout and a text locator for the `Skip` button:

```suggestion
		const isNuxWelcomeDialogDisplayed = await driverHelper.isElementEventuallyLocatedAndVisible(
			this.driver,
			By.css( '.wpcom-editor-welcome-tour-frame' ),
			1000
		);

		if ( isNuxWelcomeDialogDisplayed ) {
			await driverHelper.clickWhenClickable(
				this.driver,
				driverHelper.createTextLocator( By.css( 'button' ), 'Skip' )
			);
		}
```

Reply:
I had tried `isElementEventuallyLocatedAndVisible`, which also waits for the element AFAIK) and it caused the TC job to timeout.

I think due to how often it runs (since it's part of the `Expect` method for the editor component), sometimes it might not find the dialog (I don't know the criteria for when it should be shown, exactly) and it will wait for the full timeout, adding considerably to the total test run time.

Reply:
If you run the `isElementEventuallyLocatedAndVisible` without specifying the timeout value it will use the default one, which is 30 seconds AFAIR. That might have been causing the timeouts for you, as the helper itself doesn't _eventyally_ throw, just return a bool.

Reply:
Oh wow, 30 secs as a default value for the timeout? 😱

> That might have been causing the timeouts for you, as the helper itself doesn't eventually throw, only return a bool.

But if the element is nowhere to be found and will never appear, then it'll wait until the end of the timeout. Even if we add 1s, it might add up, no?

There might be cheaper way of checking if the dialog will be shown, maybe a setting in localstorage we can check? 🤔 

That being said, It wouldn't hurt to try this approach and see how it works in practice 👍🏻

Reply:
> Oh wow, 30 secs as a default value for the timeout? 😱

Yeah, it's the default timeout for all the waiter helpers. I agree it doesn't make much sense for this one, though! 😄 

> But if the element is nowhere to be found and will never appear, then it'll wait until the end of the timeout. Even if we add 1s, it might add up, no?

Yes, so if you think it's OK (stability-wise) to perform an immediate check you can use that helper with 0s timeout. It will basically do the same as what you've written.

> There might be cheaper way of checking if the dialog will be shown, maybe a setting in localstorage we can check? 🤔

If that value is initially available then yeah, it would definitely be cheaper! You could use the `executeScript` method there:

```js
const willNuxWelcomeDialogDisplayed = await driver.executeScript( 'return localStorage.foobar' );
```

Reply:
Thanks! I'll see if there's a setting client-side that could be used to check for that.

Reply:
TIL about `driverHelper.createTextLocator( By.css( 'button' ), 'Skip' )`, nice 👍🏻

Reply:
I've decided against checking the `localStorage` or `wp.data` because we'd be tying the test with implementation details (for example, the relevant store already changed its name from `automattic/nux` to `automattic/wpcom-style-guide`, see https://github.com/Automattic/wp-calypso/commit/1cdfe2e3db96914cb798a42555d391c669803ca4#diff-c3357bc81264c87448e2e258f0c663921b260c27ecd6d5b36dc3fb6f1e7c6760). Also, the bool value that defines if it's going to be displayed is based on a REST API call response, see https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/block-editor-nux.js#L39. 

Considering it's working as is, I'll just add the 1s wait time as you suggested to decrease the likelihood of flakiness due to race conditions.